src/homepageExperience/components/steps/python/WriteDataSql.tsx

```diff
@@ -0,0 +1,283 @@
+import React, {useContext, useEffect, useState} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+import {
+  Columns,
+  ComponentSize,
+  Grid,
+  InfluxColors,
+  Panel,
+  Table,
+} from '@influxdata/clockface'
+
+import {SafeBlankLink} from 'src/utils/SafeBlankLink'
+import WriteDataHelperBuckets from 'src/writeData/components/WriteDataHelperBuckets'
+import CodeSnippet from 'src/shared/components/CodeSnippet'
+import {WriteDataDetailsContext} from 'src/writeData/components/WriteDataDetailsContext'
+
+import {getOrg} from 'src/organizations/selectors'
+import DataListening from 'src/homepageExperience/components/DataListening'
+import {getBuckets} from 'src/buckets/actions/thunks'
+import {event} from 'src/cloud/utils/reporting'
+
+const logCopyCodeSnippet = () => {
+  event('firstMile.pythonWizard.writeData.code.copied')
+}
+
+const logDocsOpened = () => {
+  event('firstMile.pythonWizard.writeData.docs.opened')
+}
+
+type OwnProps = {
+  onSelectBucket: (bucketName: string) => void
+}
+
+export const WriteDataSqlComponent = (props: OwnProps) => {
+  const org = useSelector(getOrg)
+  const dispatch = useDispatch()
+  const {onSelectBucket} = props
+
+  useEffect(() => {
+    dispatch(getBuckets())
+  }, [dispatch])
+
+  const {bucket} = useContext(WriteDataDetailsContext)
+
+  const [selectedBucket, setSelectedBucket] = useState(bucket)
+
+  useEffect(() => {
+    setSelectedBucket(bucket)
+    onSelectBucket(bucket.name)
+  }, [bucket, onSelectBucket])
+
```

Comment:
Same comment here re - I might have the code snippets exported from another file so that they're not mixed in with the JSX.

Reply:
This was in keeping with the existing pattern. We can refactor though to move them.